lib/childprocess.rb

```diff
@@ -116,14 +116,8 @@ def arch
         host_cpu = RbConfig::CONFIG['host_cpu'].downcase
         case host_cpu
         when /i[3456]86/
-
-          # Ruby 2.4 unifies Bignum and Fixnum into Integer. Also, I've heard that Darwin no
-          # longer has this issue, so on newer Ruby/Darwin combos this check shouldn't be
-          # needed anyway. Leaving it here for older combinations, however.
-
-          # Check for Ruby older version of Ruby
-          if (RUBY_VERSION < '2.4') && (os == :macosx) && (0xfee1deadbeef.is_a?(Fixnum))
-            # Darwin always reports i686, even when running in 64bit mod
+          if os == :macosx && RUBY_VERSION < '2.4' && is_64_bit?
```

Comment:
Hmm. That's not exactly what this says, but you might be right anyway. Let me push a small change to the code to make it read more clearly on what it means, and then we can discuss if you want the check?